From b5a2e72638b725d88e5af0dd2e69bb62e5e54482 Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sun, 16 Aug 2026 20:40:24 +0200
Subject: [PATCH 1/2] Only include headers for JACK and ALSA when not compiling
 the LV2 plugin

Make sure to only include `MusicIO/JackEngine.h` when
`YOSHIMI_LV2_PLUGIN` is not defined.
Make sure to only include `MusicIO/AlsaEngine.h` if `YOSHIMI_LV2_PLUGIN`
is not defined and `HAVE_ALSA` is defined.
Always include `MusicIO/MusicIO.h` as it is always required for e.g.
`SinglethreadedBeatTracker`.
Always include `memory` because it is always required when using
`std::make_shared`.

Fixes: https://github.com/Yoshimi/yoshimi/issues/250
Signed-off-by: David Runge <dave@sleepmap.de>
---
 src/MusicIO/MusicClient.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/MusicIO/MusicClient.cpp b/src/MusicIO/MusicClient.cpp
index 71097e76..70a974df 100644
--- a/src/MusicIO/MusicClient.cpp
+++ b/src/MusicIO/MusicClient.cpp
@@ -22,12 +22,18 @@
 */
 
 #include "MusicIO/MusicClient.h"
+#include "MusicIO/MusicIO.h"
 #include "Misc/SynthEngine.h"
-#include "MusicIO/AlsaEngine.h"
+#ifndef YOSHIMI_LV2_PLUGIN
 #include "MusicIO/JackEngine.h"
+#ifdef HAVE_ALSA
+#include "MusicIO/AlsaEngine.h"
+#endif // HAVE_ALSA
+#endif // YOSHIMI_LV2_PLUGIN != 1
 #include <iostream>
 #include <stdlib.h>
 #include <cassert>
+#include <memory>
 #include <thread>
 #include <string>
 #include <set>
-- 
2.55.0


From 676a4c23bd2b12935dea01d50ca41626ee1b1eed Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sun, 16 Aug 2026 20:49:33 +0200
Subject: [PATCH 2/2] Remove unused includes of `iostream` and `set` from
 `MusicClient.cpp`

Signed-off-by: David Runge <dave@sleepmap.de>
---
 src/MusicIO/MusicClient.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/MusicIO/MusicClient.cpp b/src/MusicIO/MusicClient.cpp
index 70a974df..83f04908 100644
--- a/src/MusicIO/MusicClient.cpp
+++ b/src/MusicIO/MusicClient.cpp
@@ -30,13 +30,11 @@
 #include "MusicIO/AlsaEngine.h"
 #endif // HAVE_ALSA
 #endif // YOSHIMI_LV2_PLUGIN != 1
-#include <iostream>
 #include <stdlib.h>
 #include <cassert>
 #include <memory>
 #include <thread>
 #include <string>
-#include <set>
 
 using std::string;
 using std::unique_ptr;
-- 
2.55.0

